client/state/notification-settings/actions.js

```diff
@@ -24,3 +39,90 @@ export const updateNotificationSettings = settings => ( {
 	type: NOTIFICATION_SETTINGS_UPDATE,
 	settings,
 } );
+
+export const toggle = ( source, stream, setting ) => dispatch => {
+	dispatch( {
+		type: NOTIFICATION_SETTINGS_TOGGLE_SETTING,
+		source,
+		stream,
+		setting,
+	} );
+};
+
+export const toggleWPcomEmailSetting = setting => toggle( 'wpcom', 'email', setting );
+
+export const fetchSettings = () => dispatch => {
+	dispatch( { type: NOTIFICATION_SETTINGS_FETCH } );
+
+	wpcom
+		.undocumented()
+		.me()
+		.getNotificationSettings( ( error, data ) => {
+			const type = error
+				? NOTIFICATION_SETTINGS_FETCH_FAILED
+				: NOTIFICATION_SETTINGS_FETCH_COMPLETE;
+
+			dispatch( {
+				type,
+				error,
+				data,
+			} );
+		} );
+};
+
+function buildSavePayload( source, settings ) {
+	switch ( source ) {
+		case 'wpcom':
+			return {
+				wpcom: settings,
+			};
+		case 'other':
+			return {
+				other: settings,
+			};
+		default:
```

Comment:
Use `case: 'blogs'` here. The `default` case looks like some past mistake that should be fixed.

Reply:
I thought so too and I've made some note of this. I haven't purposely tested that change yet though but I'll update it once I have :)